code/datums/elements/selfknockback.dm

```diff
@@ -0,0 +1,32 @@
+/datum/element/selfknockback
+	element_flags = ELEMENT_DETACH
```

Comment:
You don't need this flag, signals get cleaned up automaticaly in qdel. Keep the code in Detach though